pkg/operator/controller/ingress/controller.go

```diff
@@ -713,6 +720,56 @@ func (r *reconciler) ensureIngressDeleted(ingress *operatorv1.IngressController)
 	return utilerrors.NewAggregate(errs)
 }
 
+// GCS: findMostRecentIngress and findCondition copied from openshift-router:status.go...curious if there is a better way
+
+// findMostRecentIngress returns the name of the ingress status with the most recent Admitted condition transition time,
+// or an empty string if no such ingress exists.
+func (r *reconciler) clearRouteStatus(icName string) error {
+	routeList := &routev1.RouteList{}
+
+	// List all routes.
+
+	start := time.Now()
+	if err := r.client.List(context.TODO(), routeList); err != nil {
+		return fmt.Errorf("failed to list all routes in order to clear route status")
+	}
+	elapsed := time.Since(start)
+	fmt.Println(elapsed)
+
+	// Go through each route and clear status if admitted by this ingress controller.
+	start2 := time.Now()
+	for _, route := range routeList.Items {
+		updated := route.DeepCopy()
+		for i := range route.Status.Ingress {
+			if condition := findCondition(&route.Status.Ingress[i], routev1.RouteAdmitted); condition != nil {
+				if routerName := &route.Status.Ingress[i].RouterName; routerName != nil && *routerName == icName {
+					// Remove this status since it matches our routerName.
+					updated.Status.Ingress = append(route.Status.Ingress[:i], route.Status.Ingress[i+1:]...)
+					if err := r.client.Update(context.TODO(), updated); err != nil {
+						return fmt.Errorf("failed to clear route status for %s with routerName %s: %v", route.Name, icName, err)
+					}
+					continue
+				}
+			}
+		}
+	}
+	elapsed2 := time.Since(start2)
+	fmt.Println(elapsed2)
```

Comment:
This timing information could be quite useful.  Can you make these print statements into formal log messages?

Reply:
Sure